src/goto-programs/name_mangler.h

```diff
@@ -0,0 +1,155 @@
+/// \file name_mangler.h
+/// \brief Mangle names of file-local functions to make them unique
+/// \author Kareem Khazem <karkhaz@karkhaz.com>
+
+#ifndef CPROVER_GOTO_PROGRAMS_NAME_MANGLER_H
+#define CPROVER_GOTO_PROGRAMS_NAME_MANGLER_H
+
+#include <util/message.h>
+#include <util/rename_symbol.h>
+
+#include "goto_model.h"
+
+#include <forward_list>
+#include <regex>
+
+/// \brief Mangles the names in an entire program and its symbol table
+///
+/// The type parameter to this class should be a functor that has a no-arg
+/// constructor and an `operator()` override with the following signature:
+///
+///     irep_idt operator()(const symbolt &, const std::string &);
+///
+/// The return type doesn't actually have to be an irep_idt, just something
+/// that can be assigned to one. The function is expected to return the
+/// mangled name of its \ref symbolt argument, incorporating the second
+/// argument into the mangled name if possible.
+template <class MangleFun>
+class name_manglert
```

Comment:
Suggest `function_name_manglert`, as I supposed it really mangled all symbols at first